coral-common/src/main/java/com/linkedin/coral/common/transformers/OperatorTransformer.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2017-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.transformers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlSelect;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+
+/**
+ * Abstract class for generic transformations on SqlNodes
+ */
+public abstract class OperatorTransformer {
```

Comment:
nit: Not sure about the naming here. This is a SqlCallTranformer, which can modify any aspect of a SqlCall, not just the operator.
Will get back with some naming ideas.

Reply:
+1 to the name change. I was waiting for the investigation on `SqlBasicCall` as well to suggest names.